### PR TITLE
Data+Model

### PR DESCRIPTION
**Pull request by isamu-isozaki**
Work in development. Don't merge. I'm using the titanic dataset. In this branch, I'll do
1. Clean data so can be loaded into the model.
2. Make a model that takes m as an input and outputs a single value: the prediction.

### CODE REVIEW

**Comment by ethanmoyer** [comment]
Can you create a function too (I.e. run_model(n)) that accepts n units as a parameter and returns the accuracy. Also, could you implement early stopping with a default parameter in the function definition?

**Comment by isamu-isozaki** [comment]
@ethanmoyer Gotcha. Will try to finish that tomorrow! Thanks for the comment.

**Comment by isamu-isozaki** [comment]
@ethanmoyer Should be done. You can check.

**Comment by isamu-isozaki** [comment]
Opened pull request.

File: binary_search_networks/test.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Author: Isamu Isozaki
+# Date: 2020/11/10
+# Purpose: Take X, y and model as input and returns test accuracy
+
+def test(X, y, model):
+    _, accuracy = model.evaluate(X, y)
```
Comment (ya332) [inline binary_search_networks/test.py:13]: Can you get accuracy, precision, recall, f1-score, and the confusion matrix as well?

**Review by ya332 — CHANGES_REQUESTED** [review]

File: binary_search_networks/test.py (hunk 1 above)
Comment (isamu-isozaki) [reply binary_search_networks/test.py:13]: Yup. I'll do that in train.py too hopefully by the end of today.

Comment (ya332) [reply binary_search_networks/test.py:13]: Here is how I did it for mine with `scikit-learn==0.20.4`. Feel free to modularize. 
```Python3
print("Start evalution of the model")

# Evaluting the model with different metrics
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score
from sklearn.metrics import auc
from sklearn.metrics import classification_report
from sklearn.metrics import roc_curve

# print(X_train.shape, X_test.shape, y_pred.shape, y_train.shape, y_test.shape)
acc_score = accuracy_score(y_test, y_pred)
fpr, tpr, thresholds = roc_curve(y_test, y_pred) 
area_under_curve = auc(fpr, tpr)
cr = classification_report(y_test, y_pred)
cm = confusion_matrix(y_test, y_pred)

print("Evaluation is over.")
print("#"*30)
print("Model Metrics")
print("Accuracy Score: ", acc_score)
print("Area Under Curve: ", area_under_curve)
print("Classification Report:\n", cr)
print("Confusion Matrix:\n", cm)
print("#"*30)
```

**Comment by isamu-isozaki** [comment]
Added precision, recall, f1 score calculation for the test, and some useful comments for people who are new to ml.

File: binary_search_networks/test.py (hunk 1 above)
Comment (isamu-isozaki) [reply binary_search_networks/test.py:13]: Thank you! Referenced your code a bunch.

**Review by ya332 — APPROVED** [review]
LGTM!
